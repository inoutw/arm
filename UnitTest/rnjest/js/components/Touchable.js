import React, { Component } from 'react';
import { Text, TouchableWithoutFeedback, View } from 'react-native';

export default class Touchable extends Component {
	onRowPress() {
		this.myCoolFunction();
	}

	myCoolFunction() {
		console.log('hi');
	}

	render() {
		return (
			<TouchableWithoutFeedback onPress={this.onRowPress.bind(this)} testID={'toucha'}>
				<View>
					<Text>Hello World</Text>
				</View>
			</TouchableWithoutFeedback>
		);
	}
}