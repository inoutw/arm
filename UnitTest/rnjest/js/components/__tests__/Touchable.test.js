
import React from 'react';
import renderer from 'react-test-renderer';
import Touchable from '../Touchable';

jest.mock('TouchableWithoutFeedback', () => {
	const mockComponent = require('jest-react-native');
	return mockComponent('TouchableWithoutFeedback');
});
it('handle press on Touchable', () => {

})