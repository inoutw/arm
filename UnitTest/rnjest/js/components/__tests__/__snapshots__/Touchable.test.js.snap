// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle press on Touchable 1`] = `
<TouchableWithoutFeedback
  onPress={[Function]}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Hello World
    </Text>
  </View>
</TouchableWithoutFeedback>
`;
